core/src/main/java/com/linecorp/armeria/client/circuitbreaker/NonBlockingCircuitBreaker.java

```diff
@@ -141,17 +147,25 @@ public boolean canRequest() {
         return true;
     }
 
-    private State newOpenState() {
-        return new State(CircuitState.OPEN, config.circuitOpenWindow(), NoOpCounter.INSTANCE);
+    @Override
+    public CircuitBreakerState state() {
+        return state.get();
+    }
```

Comment:
Currently a `CircuitBreakerState` transitions to another state, a new object is allocated that users don't want to.
Users try to keep track of the current state of a `CircuitBreaker` by the returned `CircuitBreakerState`.
If we allocate a new object, users can't see the updated state with the old `CircuitBreakerState`.

Would you mind refactoring `NonBlockingCircuitBreaker` to have a uniq `CircuitBreakerState` and update the new `CircuitState`, `EventCounter` and `timedOutTimeNanos` internally?
```java
class DefaultCircuitBreakerState implement CircuitBreakerState {

  private volatile CircuitBreakerState delegate;

  void updateState(CircuitBreakerState newState) {
       // Atomically update delegate using compareAndSet()
  }

}
```

Reply:
Thanks for good suggestion! I want to handle it after discussing from https://github.com/line/armeria/pull/4012#discussion_r785420211

Reply:
Q) Did we decide what do with this method? (revert vs. `CircuitState` vs. refine `CircuitBreakerState` and make it public)

Personally, I feel like exposing the internal state may make future updates difficult 🤔

Reply:
We decided to revert `CircuitBreakerState` and address the original idea from Trustin.
https://github.com/line/armeria/pull/4012#discussion_r785759625

Reply:
I misunderstood it. I'v updated like below 

- Deprecate `canRequest`
- Add `tryRequest` renamed from `canRequest`
- Add `circuitState` method to return current `circuitState`